src/psyclone/domain/gocean/kernel/psyir.py

```diff
@@ -504,15 +519,20 @@ def procedure_name(self):
         :returns: the kernel procedure name specified by the metadata.
         :rtype: str
         '''
-        return self._code
+        return self._procedure_name
 
     @procedure_name.setter
     def procedure_name(self, value):
         '''
         :param str value: set the procedure name specified in the \
             metadata to the specified value.
         '''
-        self._code = value
+        pattern = re.compile(r'[a-zA-Z_][\w]*')
```

Comment:
Again, if we put this pattern as an instance variable then we could compile it just once and reuse it here and in the `name` setter.

Reply:
Done.